File: c program/queue.c
#include <stdio.h>

int q[5];

int f = -1, r = -1;

void enqueue();
void dequeue();
void traverse();

int main()
{
    int ch;

    while (1)
    {
        printf("enter the operation");
        scanf("%d", &ch);

        switch (ch)
        {
        case 1:
            enqueue();
            break;

        case 2:
            dequeue();
            break;

        case 3:
            traverse();
            break;

        case 4:
            printf("exiting");
            break;

        default:
            printf("invalid choice");

            break;
        }
    }
    return 0;
}

void enqueue()
{
    int v;
    printf("enter the value of for inserting");
    scanf("%d", &v);
    if (f == 4)
    {
        printf("queue is full");
    }
    else
    {
        q[f] = v;
        f++;
    }
}

void dequeue()
{
    if (f == -1)
    {
        printf("queue is empty");
    }
    else
    {
       
        printf("%d", q[r]);
         r++;
    }
}

void traverse()
{
    if (f == -1)
    {
        printf("queue is empty");
    }
    else
    {
        for (int i = r; i <= f; i++)
        {
            printf("%d", q[i]);
        }
    }
}

File: dsa/queue.py
size=int(input("values"))
print("for inserting 1","\n","for delete 2","\n","for display 3","\n","for exiting 4")


f=-1
r=-1

q=[]

def insert():
    global f
    global r
    if (f==size):
        print("queue is full")
    

    else:
        if (f==r):
            f=-1
            r=-1
        
        n=int(input("value for insertion"))
        f=f+1
        q.append(n)
        if (r==-1):
            r=r+1

def delete():
    global r
    if(f==-1):
        print("queue is empty")
    
    else:
        print("deleted element is",q[r])
        r=r+1

def display():
      if(f==-1):
        print("queue is empty")
      else:
        for i in range(f+1):
            print(q[i])

while(1):
    
    ch=int(input(""))
    if (ch==1):
        insert()
    
    elif (ch==2):
        delete()

    elif (ch==3):
        display()   
    elif (ch==4):
        break
    else:
        print("wrong choice")
